src/components/pages/MentorMenteeMatching/MentorsCard.js

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react';
+import { Card, Select } from 'antd';
+import { EditOutlined } from '@ant-design/icons';
+import { options, empty } from './MentorOptions';
+
+const MentorsCard = ({ data }) => {
+  const [edit, setEdit] = useState(false);
+
+  const [mentors, setMentors] = useState({
+    first: '',
+    second: '',
+    third: '',
+  });
+
+  return (
+    <>
+      <Card
+        title="Mentors"
+        style={{
+          width: '30%',
+        }}
+        extra={
+          !mentors.first && !mentors.second && !mentors.third ? (
+            ''
+          ) : edit ? (
+            <p
+              onClick={() => setEdit(!edit)}
+              style={{
+                textDecoration: 'underline',
+                cursor: 'pointer',
+              }}
+            >
+              save
+            </p>
+          ) : (
+            <EditOutlined
+              style={{ marginTop: '2px' }}
+              onClick={() => setEdit(!edit)}
+            />
+          )
+        }
+      >
+        {edit || empty(mentors.first) ? (
+          <div style={{ width: '80%' }}>
+            <Select
+              name="first"
+              onChange={value => setMentors({ ...mentors, first: value })}
+              showSearch
+              style={{ width: '100%' }}
+              optionFilterProp="children"
+              value={mentors.first || 'Assign Mentor'}
+            >
+              {options(data)}
+            </Select>
+          </div>
+        ) : (
+          <div style={{ width: '80%' }}>
+            <p style={{ marginLeft: '5px' }}>{mentors.first}</p>
+          </div>
+        )}
+
+        {edit || empty(mentors.second) ? (
+          <div style={{ width: '80%', marginTop: '20px' }}>
+            <Select
```

Comment:
Why is Select reused this way? These <Select /> components look very similar and could be modularized with a single prop to make the code a little more DRY.

Reply:
Yes, I spent a long time trying to come up with a way to make one select input and have the rest be props, unfortunately doing that created more problems than it solved. ie, functionality would only be available for one input box rather than all. I was able to get around that but it created another issue, which was the use of props was becoming really hectic. I decided to go this route because the functionality is important and since we are using dummy data for now, I did not want to confuse future developers with unnecessary issues.